src/ThemeProvider.tsx

```diff
@@ -2,31 +2,51 @@ import PropTypes from 'prop-types';
 import * as React from 'react';
 import { useContext, useMemo } from 'react';
 
-export interface ThemeProviderProps {
-  prefixes: Record<string, unknown>;
+export interface ThemeContextValue {
+  prefixes: Record<string, string>;
+  dir?: string;
 }
 
-const ThemeContext = React.createContext({});
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ThemeProviderProps extends Partial<ThemeContextValue> {}
+
+const ThemeContext = React.createContext<ThemeContextValue>({ prefixes: {} });
 const { Consumer, Provider } = ThemeContext;
 
-function ThemeProvider({ prefixes, children }) {
-  const copiedPrefixes = useMemo(() => ({ ...prefixes }), [prefixes]);
+const ThemeProvider: React.FC<Partial<ThemeContextValue>> = ({
+  prefixes = {},
+  dir,
+  children,
+}) => {
+  const contextValue = useMemo(
+    () => ({
+      prefixes: { ...prefixes },
+      dir,
+    }),
+    [prefixes, dir],
+  );
 
-  return <Provider value={copiedPrefixes}>{children}</Provider>;
-}
+  return <Provider value={contextValue}>{children}</Provider>;
+};
 
 ThemeProvider.propTypes = {
-  prefixes: PropTypes.object.isRequired,
-};
+  prefixes: PropTypes.object,
+  dir: PropTypes.string,
+} as any;
 
 export function useBootstrapPrefix(
   prefix: string | undefined,
   defaultPrefix: string,
 ): string {
-  const prefixes = useContext(ThemeContext);
+  const { prefixes } = useContext(ThemeContext);
   return prefix || prefixes[defaultPrefix] || defaultPrefix;
 }
 
+export function useRTL() {
+  const { dir } = useContext(ThemeContext);
+  return dir === 'rtl';
```

Comment:
What alternatives to storing this in context have we considered? This adds a new RB specific api required in addition to the upstream one. What about stealing their JS helper? https://github.com/twbs/bootstrap/blob/main/js/src/util/index.js#L230 

If we want a reactive hook still could combine with a mutationObserver to trigger updates, though i'm not sure how common dynamically changing the read direction is aside from demonstrations in docs

Reply:
I used the context to support SSR since `document` wasn't available.  Wasn't sure if there was another way?

Reply:
hmm you are right. what if we made this a function? presumably this is actually only relevant on the client, but we'd need to defer accesing the document til after render...or maybe set `dir` state in a useEffect? Do we think there is a reason why we'd need this during a SSR run, i suspect most usage is in event handlers.

Reply:
From my list before, the components that need RTL changes are dropdown, modal, carousel, tooltip.  I think you're right that these aren't really needed during SSR - will have to double check.

Reply:
@jquense looking at upstream, this is where RTL is used:
- Modal - setting padding for scrollbar when modal is shown
- Dropdown - menu placement
- Tooltip/Popover - tooltip/popover placement
- Carousel - keyboard control

Seems like we would need SSR support for dropdowns with renderOnMount or tooltips/popovers with show=true?